el/index.js

```diff
@@ -93,6 +106,28 @@ module.exports = yeoman.generators.Base.extend({
       this.writeFileFromString(file, 'app/elements/elements.html');
     }
 
+    // create an associated test file
+    if (this.testType && this.testType !== 'None') {
+      var testDir = 'app/test';	
+      
+      // Copy approprate template for the chosen test style     
+      if (this.testType === 'TDD') {
+        this.template(path.join(__dirname, 'templates/test/_tdd.html'), path.join(testDir, this.elementName+'-basic.html'));
+      } else if (this.testType === 'BDD') {
+        this.template(path.join(__dirname, 'templates/test/_bdd.html'), path.join(testDir, this.elementName+'-basic.html'));
+      }
+
+      // Open index.html, locate where to insert text, insert ", x-foo.html" into the array of components to test
+      var indexFileName = 'app/test/index.html';
+      var file = this.readFileAsString(indexFileName);
+      var regex = /WCT\.loadSuites\(\[([^\]]*)/;
+      var match = regex.exec(file);
+      var indexOfInsertion = match.index + match[0].length;
+      var comma = (match[1].length === 0) ? "" : ", ";
+      var newFile = file.slice(0, indexOfInsertion) + comma + "'"+this.elementName+"-basic.html'" + file.slice(indexOfInsertion);
```

Comment:
Just a thought... would it be less work to capture the entire array in the index.html, use JSON.parse to convert it, push in the file that you're adding, and write that Array back, instead of needing to try to figure out if there's a command slice it up?